castor-service/src/test/java/io/carbynestack/castor/service/persistence/cache/ReservationCachingServiceIT.java

```diff
@@ -158,4 +189,74 @@ public void givenSuccessfulRequest_whenForgetReservation_thenRemoveFromCache() {
     reservationCachingService.forgetReservation(testReservation.getReservationId());
     assertNull(reservationCache.get(testReservation.getReservationId(), Reservation.class));
   }
+
+  @Test
+  public void whenReferencedSequenceIsSplitInFragments_whenApplyReservation_thenApplyAccordingly() {
+    UUID tupleChunkId = UUID.fromString("3fd7eaf7-cda3-4384-8d86-2c43450cbe63");
+    long requestedStartIndex = 42;
+    long requestedLength = 21;
+    ReservationElement re =
+        new ReservationElement(tupleChunkId, requestedLength, requestedStartIndex);
+    String reservationId = "testReservation";
+    TupleType tupleType = MULTIPLICATION_TRIPLE_GFP;
+    Reservation r = new Reservation(reservationId, tupleType, singletonList(re));
+
+    TupleChunkFragmentEntity fragmentBefore =
+        TupleChunkFragmentEntity.of(
+            tupleChunkId, tupleType, 0, requestedStartIndex, UNLOCKED, null);
+    TupleChunkFragmentEntity fragmentPart1 =
```

Comment:
Would rather use `fragmentNew` and `fragmentRest` here.

Reply:
I think this would reflect the intention as the purpose of the test is to verify that 
- the `applyReservation` leaves fragments that reference sections _before_ the actual tuples of interest untouched
- applies the reservation on multiple fragments if the tuples of interest are actually described by / split to multiple consecutive fragments

example:
[Fragment FA withs Tuple T0-T3] [Fragment FB with Tuples T4-T5] [Fragment FC with Tuple T6]
The reservation is to be applied on tuples T4-T6
expected result:
- First fragment FA, which describes a section before the first tuple of interest, remains untouched
- Reservation is applied on Fragment FB and FC